src/Data/ProgrammesDb/EntityRepository/VersionRepository.php

```diff
@@ -140,6 +140,30 @@ public function findAvailableByProgrammeItem(string $programmeDbId): array
         return $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
+    public function findLinkedVersionsForProgrammeItem(string $programmeItemDbId): array
+    {
+        $qb = $this->getEntityManager()->createQueryBuilder()
+            ->select([
+                'p', 'streamableVersion', 'streamableVersionTypes', 'downloadableVersion', 'downloadableVersionTypes',
+                'canonicalVersion', 'canonicalVersionTypes',
+                'masterBrand', 'competitionWarning', 'competitionWarningProgrammeItem',
+            ])
+            ->from('ProgrammesPagesService:ProgrammeItem', 'p')
+            ->leftJoin('p.masterBrand', 'masterBrand')
+            ->leftJoin('masterBrand.competitionWarning', 'competitionWarning')
+            ->leftJoin('competitionWarning.programmeItem', 'competitionWarningProgrammeItem')
+            ->leftJoin('p.downloadableVersion', 'downloadableVersion')
+            ->leftJoin('downloadableVersion.versionTypes', 'downloadableVersionTypes')
+            ->leftJoin('p.streamableVersion', 'streamableVersion')
+            ->leftJoin('streamableVersion.versionTypes', 'streamableVersionTypes')
+            ->leftJoin('p.canonicalVersion', 'canonicalVersion')
+            ->leftJoin('canonicalVersion.versionTypes', 'canonicalVersionTypes')
+            ->where('p.id = :dbId')
+            ->setParameter('dbId', $programmeItemDbId);
+
+        return $qb->getQuery()->getOneOrNullResult(Query::HYDRATE_ARRAY);
+    }
```

Comment:
Please extract the query and run it with EXPLAIN to make sure we use indexes and it perform as expected

Reply:
It takes 100ms. From my point of view is lightly slow, but I don't see either any problems with indexes.

Reply:
Done